alephium/caip19.md

```diff
@@ -0,0 +1,46 @@
+---
+namespace-identifier: alephium-caip19
+title: Alephium Namespace - Assets
+author: Hongchao Liu (@h0ngcha0)
+discussions-to: https://github.com/ChainAgnostic/namespaces/pull/117
+status: Draft
+type: Standard
+created: 2024-06-20
+updated: 2024-06-20
+requires: ["CAIP-2", "CAIP-19"]
+---
+
+## Rationale
+
+Tokens are first class citizens on Alephium. Just like the native
+token `ALPH`, all tokens on Alephium are managed by UTXOs, which is
+battle tested for secure asset management.
+
+New tokens are deployed through the deployment of new contracts, with
+token's id matching the id of the issuing contract. The process of
+token issuance is the same for both fungible and non-fungible tokens.
+
+## Specification of token id
+
+Token's id is the same as its issuing contract's id, which is a 32
+bytes blake2b hash encoded as a hex string.
+
+## Syntax
+
+Alephium token id represented as 32 byte array encoded as hex string.
+
+## Examples
+
+```
+383bc735a4de6722af80546ec9eeb3cff508f2f68e97da19489ce69f3e703200
```

Comment:
Here too I'm confused about the missing `alephium:` prefix and CAIP-2 segment!

Reply:
Tokens are "universal" as well in the sense that they are not tied to a specified network. It should probably be something like `alephium:universal:{tokenId}`, but I am not sure about how to reconcile with the `CAIP-2` part in this case, similar to address.

Reply:
yeah, that makes sense! if something is shard-specific or chain-specific it would go in the `chainId` reference (the second segment), and if you want to create a special case for `universal` addresses, you just add it to the caip2 profile "specification" section!

Reply:
Thanks! I added the special case for `universal` in this [commit](https://github.com/ChainAgnostic/namespaces/pull/117/commits/d128daedf783e31eef823ee20259c85fd77977f4)